### ISSUE TITLE
Can't jump to the function call when ctrl click on a function definition in VSCode Bash IDE plugin

### ISSUE THREAD

**Issue by YevheniiPokhvalii** [bash-lsp/bash-language-server#515]
I can jump/go to a function Definition﻿ by clicking `ctrl+left mouse` or `f12` on a function call, but it does not seem to work backwards - I can't jump from a function Definition to the function call, like it is implemented, for example, in the built-in plugin [Shell Script](https://plugins.jetbrains.com/plugin/13122-shell-script) for IntelliJ IDEA.
Is this a current VSCode Bash IDE plugin/bash-language-server limitation or am I doing something wrong?
I am using the latest version of https://marketplace.visualstudio.com/items?itemName=mads-hartmann.bash-ide-vscode

**Comment by skovhus** [comment]
I can't really reproduce this. If I check right click on `add_a_user` and select `Go to references` it correctly shows the 3 calls.

Let me know if you have a reproducible example.

```sh
#!/bin/sh

# Helper function to add a user
add_a_user()
{
  USER=$1
  PASSWORD=$2
  shift; shift;
  # Having shifted twice, the rest is now comments ...
  COMMENTS=$@
  echo "Adding user $USER ..."
  echo useradd -c "$COMMENTS" $USER
  echo passwd $USER $PASSWORD
  echo "Added user $USER ($COMMENTS) with pass $PASSWORD"
}

###
# Main body of script starts here
###
echo "Start of script..."
add_a_user bob letmein Bob Holness the presenter
add_a_user fred badpassword Fred Durst the singer
add_a_user bilko worsepassword Sgt. Bilko the role model
echo "End of script..."

```

**Comment by YevheniiPokhvalii** [comment]
> I can't really reproduce this. If I check right click on `add_a_user` and select `Go to references` it correctly shows the 3 calls.
> 
> Let me know if you have a reproducible example.
> 
> ```shell
> #!/bin/sh
> 
> # Helper function to add a user
> add_a_user()
> {
>   USER=$1
>   PASSWORD=$2
>   shift; shift;
>   # Having shifted twice, the rest is now comments ...
>   COMMENTS=$@
>   echo "Adding user $USER ..."
>   echo useradd -c "$COMMENTS" $USER
>   echo passwd $USER $PASSWORD
>   echo "Added user $USER ($COMMENTS) with pass $PASSWORD"
> }
> 
> ###
> # Main body of script starts here
> ###
> echo "Start of script..."
> add_a_user bob letmein Bob Holness the presenter
> add_a_user fred badpassword Fred Durst the singer
> add_a_user bilko worsepassword Sgt. Bilko the role model
> echo "End of script..."
> ```

I mean click on **add_a_user** on the _add_a_user bob letmein Bob Holness the presenter_ line. It won't send you back to the _add_a_user()_ function.

**Comment by skovhus** [comment]
A much better heuristic for symbol resolution has been released as `bash-language-server@4.2.0` and for vscode in the latest extension 1.24.0. It opens up for implementing renaming of symbols, and I also believe this solves the issue here. 

Please let me know if it works for you or if you have any feedback to the new version.

**Comment by YevheniiPokhvalii** [comment]
Thanks! Seems to be working now.
